proto/substrait/algebra.proto

```diff
@@ -59,6 +65,31 @@ message RelCommon {
 
       substrait.extensions.AdvancedExtension advanced_extension = 10;
     }
+
+    enum ComputationType {
+      COMPUTATION_TYPE_UNSPECIFIED = 0;
+      COMPUTATION_TYPE_HASHTABLE = 1;
+      COMPUTATION_TYPE_BLOOM_FILTER = 2;
+      COMPUTATION_TYPE_UNKNOWN = 9999;
```

Comment:
The problem with any is that it's guaranteed to not work cross platforms.  With a numbering scheme it might not work cross platform but if we have enough of the computation types defined (or if there are a limited number of computations possible per relation) it's likely to work.  I would prefer we add the new computation types liberally to using UNKNOWN/CUSTOM.

Reply:
I don't see how ANY is any more/less compatible than COMPUTATION_TYPE_UNKNOWN. 

The issue I have with this in general is that nearly every time I've worked on wrt a sideband operation it has needed some custom plan metadata to work. Not having an extension point here means I can't see using this well for any of those situations. 

Can you expound on the scenario you're trying to protect against?

Reply:
Any requires knowledge of the format of the unknown data type.  The unknown type just means "count up the types that you know of that you might have and use the Xth one".  I don't need to know about how to serialize in that case, just how to count the data types I expose/desire.

Reply:
I'm ignoring any if we can solve duckdb without it.